ftp.go

```diff
@@ -144,21 +145,33 @@ func Dial(addr string, options ...DialOption) (*ServerConn, error) {
 		}
 	}
 
-	tconn, err := dialFunc("tcp", addr)
+	// Use the resolved IP address in case addr contains a domain name
+	// If we use the domain name, we might not resolve to the same IP.
+	dnsResolver := net.Resolver{
+		Dial: func(ctx context.Context, network string, address string) (net.Conn, error) {
+			return dialFunc(network, address)
+		},
+	}
+	host, port, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, err
+	}
+	addrs, err := dnsResolver.LookupIPAddr(context.Background(), host)
 	if err != nil {
 		return nil, err
 	}
 
-	// Use the resolved IP address in case addr contains a domain name
-	// If we use the domain name, we might not resolve to the same IP.
-	remoteAddr := tconn.RemoteAddr().(*net.TCPAddr)
+	tconn, err := dialFunc("tcp", fmt.Sprintf("%s:%s", addrs[0].IP.String(), port))
```

Comment:
This does not work for IPv6 addresses. Please use [JoinHostPort](https://pkg.go.dev/net#JoinHostPort)

Reply:
Updated.